### ISSUE TITLE
Let an in-app way to access background player controls

### ISSUE THREAD

**Issue by B0pol** [TeamNewPipe/NewPipe#3515]
#### Describe the feature you want

I'd like to have an in-app access to background player control.
I've thought about placing it in the three dots menu (where open in browser and settings stand), tell us if you've found a better place, or even create a pull request

#### Is your feature request related to a problem? Please describe it

On TV there is no way to access notifications. They may popup on the screen if you are on the launcher, but that's it. And there is also no way to access background player controls within the app.
#### Additional context

Hene is the screenshot of background player controls
<img src="https://raw.githubusercontent.com/TeamNewPipe/NewPipe/dev/fastlane/metadata/android/en-US/images/phoneScreenshots/shot_05.png" width="200" />
On my tv, the only way to control is to use play / pause / next / previous button, but that's really basic (i.e. I can't see the duration, changie speed…) and that may not be the case for everyone.

[This line is the notification click listener to open controls](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/player/BackgroundPlayer.java), it will surely help.

If i've understood, it will close #3324 
#### How will you/everyone benefit from this feature?

Better accessibility for people not having an (easy) access to notifications. It's unusable right now from NewPipe.

**Comment by opusforlife2** [comment]
One possible solution: #2477

**Comment by Stypox** [comment]
#2477 is the best option in my opinion, as it doesn't involve adding extra buttons here and there, and also makes queues more accrssible and customizable

**Comment by opusforlife2** [comment]
Thinking it over, one big problem I see with #2477 is that to access the background player tab, the user would have to lose their entire backstack. So there is a trade-off between accessing the player from the notification versus from the home page tab.

One possible solution to losing backstack is that #3156 is modified so that tapping the home button from the drawer _adds_ to the backstack instead of clearing it. This way, the user could access the home page tabs freely without worrying about losing their backstack.

But I'm thinking of a better alternate solution: Let the unified player PR land. It already has the code in place for a minimised player at the bottom which can be pulled up to show the video details fragment. This could instead be modified to show up minimised when the background playback button is tapped. Pulling this up should then let the user easily access the background player and their queue. This behaviour will also be intuitive to most users as music apps do things similarly.

**Comment by opusforlife2** [comment]
> But I'm thinking of a better alternate solution: Let the unified player PR land. It already has the code in place for a minimised player at the bottom which can be pulled up to show the video details fragment. This could instead be modified to show up minimised when the background playback button is tapped. Pulling this up should then let the user easily access the background player and their queue. This behaviour will also be intuitive to most users as music apps do things similarly.

Eh? Eh?

**Comment by mitosagi** [comment]
Currently, if you long press on the minimized unified player, the channel page will appear, but I think it would be better if you could access the player controls in this way instead.

**Comment by Stypox** [comment]
@mitosagi the notification is the chosen way to access the play queue. Hiding unintuitive shortcuts under long press is not a good idea imo. It would be better to instead provide a queue main page tab.

**Comment by Madis0** [comment]
With the unified player release, I think this is no longer needed (as there is no separate "background player")

**Comment by B0pol** [comment]
No, it's still needed. How are you going to control queue? How are you going to control the video in background?

**Comment by MD77MD** [comment]
i think the easiest way is to have newPipe play youtube background videos the same way it handles soundcloud tracks. when opening a SoundCloud track, you get the hole player experience including double tap to skip, progress bar ..etc. for switching back to video is easy as well, as we can have the background playback to be a dynamic button. that is to switch between background playback mode and video playback mode as follow:

1. taping background playback button will switch player to audio mode + the button will change to the video mode new icon ( new icon can have play button shape)
2. taping on video playback button will switch player to video mode + the button will change back to background icon

i guess my idea is close to what @opusforlife2  was trying to pitch but with more details.

**Comment by M1CK431** [comment]
Hi,

I'm exactly in the use case described here:
> Better accessibility for people not having an (easy) access to notifications. It's unusable right now from NewPipe.

As a SailfishOS user, **I haven't any way to see android media player notifications at all.**

I'm aligned with @opusforlife2 and @Stypox about the fact #2477 is probably the best way to implement that. However, this issue is now open since **more than one year**, despite the "good first issue" tag (aka "should be easy to implement").

And so, IMHO, it could be an acceptable solution to add a "quick and dirty" way (button, long press, double tab, whatever...) somewhere in the main app **right now** and then remove it when the ideal solution will be released.

This approach allow people like me to gain access to the full power of newpipe without having to wait months (or even years) for  a feature that seems to require a lot of work.

wdyt?

**Comment by SameenAhnaf** [comment]
>it could be an acceptable solution to add a "quick and dirty" way (button, long press, double tab, whatever...) somewhere in the main app right now and then remove it when the ideal solution will be released.

I'd highly recommend having #5235. The playback controls for background and popup players could be placed at the bottom just like current queue manager.

<details><summary> Current queue manager  </summary>

![IMG_20210803_055048](https://user-images.githubusercontent.com/79245641/127937541-5c4f1f49-95cc-417e-815e-f6cbf8a19cc9.jpg)

</summary> </details>

<details><summary> Swipe-up queue accessibility</summary>

https://user-images.githubusercontent.com/76399981/102719213-b74a2b00-42e4-11eb-800f-c216d9a4dc69.jpg

</summary> </details>

<details><summary> Swipe up queue looks </summary>

https://user-images.githubusercontent.com/76399981/102719214-b913ee80-42e4-11eb-8422-61b309058f08.jpg

</summary> </details>

**Comment by M1CK431** [comment]
Why not @SameenAhnaf , but again, that feature seems to require a lot of work and could lead to people which can't control the background player at all during months or even years :cry:

**Comment by SameenAhnaf** [comment]
@M1CK431 I understand your situation. Without notification player, queue manager is useless.

But I suggested gestures in a lot of cases and all of them were rejected by devs. 

Btw, does Newpipe preunified hide notification player as well?
https://github.com/XiangRongLin/NewPipe-preuinified

**Comment by M1CK431** [comment]
> Btw, does Newpipe preunified hide notification player as well?

Yes. :sob:

**Comment by MD77MD** [comment]
> > it could be an acceptable solution to add a "quick and dirty" way (button, long press, double tab, whatever...) somewhere in the main app right now and then remove it when the ideal solution will be released.
> 
> I'd highly recommend having #5235. The playback controls for background and popup players could be placed at the bottom just like current queue manager.
> 
> Current queue manager
> ![IMG_20210803_055048](https://user-images.githubusercontent.com/79245641/127937541-5c4f1f49-95cc-417e-815e-f6cbf8a19cc9.jpg)
> 
> Swipe-up queue accessibility
> ![IMG_20210803_070528](https://user-images.githubusercontent.com/79245641/127941757-8d563642-99e5-48c8-b9e4-85fc8ec021f4.jpg)
> 
> Swipe up queue looks
> ![IMG_20210803_070013](https://user-images.githubusercontent.com/79245641/127941461-543c327b-c554-4148-b323-95c30fb21d50.jpg)

that's awesome,  queue manger is very underrated... it
 should be accessible from everywhere...no matter what page you are in... another idea  perhaps a floating button within the app could be used as a second option  for gesture users

**Comment by Stypox** [comment]
@MD77MD use `<img src="" height="300px"/>` for screenshots, they look so big otherwise

**Comment by bmivzkrp** [comment]
Hello All.
I am a blind NewPipe user. I have been using it for more than a week. In my opinion, this problem needs at least some solution, because a blind user, guided by the experience of using other applications, believes that if he activates notifications from a particular application, the main window of this application will open. This is not the case with NewPipe. I was very surprised that I finally found the tempo control functionality and other parameters. Previously, I thought that either NewPipe does not have this functionality, or it does, but for some reason it can only be used by people who see. I often play background content because I only need audio.
The result of my impressions was this issue:
https://github.com/TeamNewPipe/NewPipe/issues/7571

**Comment by bmivzkrp** [comment]
I will add a few more words:
1. I like the opportunity to get to the player from the notification bar. But it would be nice if we could add a tooltip that would appear on the notification bar. The hint text is similar to this: "Tap the name of the content being played to go to the player". After a while, this hint will not be useful for the blind user, so he could turn it off in the settings.
2. I would like to have an alternative: quick access to the player, but this applies to a situation where the focus is on any page of NewPipe. In other cases, I m quite happy with the notification bar.

**Comment by Roy-Orbison** [comment]
It could just be an item in the main ☰ menu, like _Downloads_ is currently.

**Comment by M1CK431** [comment]
Hi @Roy-Orbison, that's an awesome idea, but I hope you are not in a hurry since this feature request is quietly heading towards his second birthday... :candle::candle:

@SameenAhnaf is there still any hope? Or is this feature request will die of old age? :trollface:

**Comment by Mrnofish** [comment]
> It could just be an item in the main ☰ menu, like Downloads is currently.

This a million times. It should be as simple as it gets and fixes a long-standing, highly commented issue that has reported time and again.

In addition to the various issues mentioned by others, the current design is weirdly opaque, it's like NewPipe is ashamed of the queue, and buries it out of view so the user has to work to find it.

**Comment by SameenAhnaf** [comment]
Guys, what about https://github.com/TeamNewPipe/NewPipe/issues/7973 instead?

**Comment by Roy-Orbison** [comment]
@SameenAhnaf As long as all the controls and progress bar were still there. I know which option sounds like less time & effort 😉.

**Comment by Mrnofish** [comment]
FWIW I think this particular issue should be addressed right now with the least effort solution mentioned above, waiting for a more impactful redesign that might require further discussion, consultation and so on, could potentially lead to another year or two of wait.

While I can't submit a PR myself, in light of all the underlying functionality being already present and invoked from somewhere else in the code, an existing dev that already has the tools and the repo, could probably add the menu option with not much more than a copy paste.

Unexciting work, sure, but a worthy QoL improvement that could probably be completed in 20 minutes while listening to a podcast.

Just my 2 cents.

**Comment by ANUKULJHA1506** [comment]
can i work on this

**Comment by M1CK431** [comment]
> can i work on this

Hoooo yes please do it! :pray: :pleading_face: 

This issue is recently celebrating his 2nd birthday, so dont hesitate even a minute :wink: 
![](https://gifimage.net/wp-content/uploads/2018/11/happy-birthday-leonardo-dicaprio-gif-4.gif)

**Comment by MD77MD** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/4405#issuecomment-1094881834
![162721972-9e7113af-1118-4a33-b46a-1bd830362ad7](https://user-images.githubusercontent.com/63132171/168490253-5b48a9c8-b7ce-4df8-ab29-6a48fa4fe5b8.png)

this could integrate a solution for this issue too... perhaps a button for the queue list could be added too

**Comment by litetex** [comment]
For another possible solution see my comment:
https://github.com/TeamNewPipe/NewPipe/pull/8424#issuecomment-1133911376

**Comment by MD77MD** [comment]
I think #8424 is the best solution that solves everyone problem with no compromises...

**Comment by peat80** [comment]
> I think #8424 is the best solution that solves everyone problem with no compromises...

The mini player is not accessible on non touch devices so this solves nothing for tv users. 🤷🏻‍♂️

**Comment by MD77MD** [comment]
not having access to the mini player is a completely different issue

**Comment by peat80** [comment]
> not having access to the mini player is a completely different issue
> 
> also dont you agree in terms of interface, what works for phone rarely works for tablets let alon on nontouch devices like tv.. what I'm trying to say is it's much better not to tie solutions accordingly... because what's best for one might not work for the other especially UI issues.

I agree, this is a nice pr that you mentioned here which solves similar issues for touch devices. But this issue here specifically mentions usage on tv. So no, this doesn’t solve everyones problems like you were saying.

**Comment by MD77MD** [comment]
> I agree, this is a nice pr that you mentioned here which solves similar issues for touch devices. But this issue here specifically mentions usage on tv. So no, this doesn’t solve everyones problems like you were saying.

😁 you are right... my mistake

**Comment by yashpalgoyal1304** [comment]
is it really a "good first issue"??

**Comment by peat80** [comment]
> Solved by #8946

@SameenAhnaf as stated earlier, this issue mentions tv users and for them it is still not solved. 🤷‍♂️

**Comment by goyalyashpal** [comment]
hi! 

doesn't the minimize player > then clicking the mini-player-control bar at the bottom reveals those?

this was added several releases ago, and works pretty well
